sfaira/data/base.py

```diff
@@ -903,6 +903,29 @@ def write_meta(
 
     # Properties:
 
+    @property
+    def id(self) -> str:
+        def clean(s):
+            return s.replace(' ', '').replace('-', '').replace('_', '').lower()
+
+        if hasattr(self, 'idx'):
+            idx = self.idx
+        else:
+            idx = "000"
```

Comment:
@davidsebfischer whenever a standard Dataset is loaded (ManyFiles or OneFile), I assume that the ID is unique even without the three digits. otherwise I don't see a way to reliable determine what three-digit idx to assign to each dataset (unless we introduce a new attribute, this would be very unintuitive to set for the person writing the dataloader as it would require counting how many dataloaders with this ID are already part of sfaira.

An comments?

Reply:
i dont think that we can guarantee the ID to be unique as it does not capture all meta data that we collect (not even our meta data infrastructure does), so i think we should keep it, the alternative would be to overload the ID which would make it less readable. In the cases ManyFiles or OneFile, I would keep the index as a function of SAMPLE_FNS OR SAMPLE_IDS, this way it s clearly defined. In this code,`idx` could be set by the DatasetGroup for example, using SAMPLE_FNS OR SAMPLE_IDS depending on whether it is ManyFiles or OneFile. `idx`  would not need to be an attribute though, that could just be handled in the constructor?

Overall, I don't see a scenario in which we need to determine the ID from the data set instance, determining it at construction using additional inputs seems fine to me.

Reply:
Yeah, I don't quite yet understand why we need this change (no ill intentions!).
What I understand:
1. We enforce a as unique as possible ID during creation
2. sfaira create requires the ID to name the Python scripts. Therefore we should not get rid of it during the creation process.
2. This PR would allow us to be lazier and not update the already existing dataloaders IDs to fit our current scheme?

Reply:
Okay, so let's agree on some terminology first.
Example of "ID": "human_placenta_2018_smartseq2_ventotormo_002_10.1038/s41586-018-0698-6"
Example of "idx": "002"

@davidsebfischer so let's get specific:
**ManyFiles or OneFile:** I define idx as a function of SAMPLE_FNS OR SAMPLE_IDS. however, as soon as we have two separate of those classes in one study, this will start producting duplicate IDs as every loader starts counting again at 1 when deriving idx from SAMPLE_FNS

**multiple files non-streamlined** the only way I can see to achieve coherent numbering of datasets in this case would be to ask the user to provide self.idx in the dataloader. this is kind of inconvenient as it will require people to check how many other datasets are already in sfaira for that study.

@Zethson 
So my intentions with this PR are as follows: even though the ID of a dataset is mostly comprised of building blocks which are already provided by the user in the dataloader, we currently require the user to manually set the ID in the dataloader. That's the part I want to get rid of (i.e. automate it). The only building block that is missing to automatically construct the ID is the idx part which we use to differentiatedataloaders which have an otherwise identical ID

Reply:
"So my intentions with this PR are as follows: even though the ID of a dataset is mostly comprised of building blocks which are already provided by the user in the dataloader, we currently require the user to manually set the ID in the dataloader. That's the part I want to get rid of (i.e. automate it). The only building block that is missing to automatically construct the ID is the idx part which we use to differentiatedataloaders which have an otherwise identical ID"

But is this really the case? When people create dataloaders with sfaira the ID is already autogenerated. Even for multiple dataloaders: https://github.com/theislab/sfaira/blob/dev/sfaira/commands/create_dataloader.py#L145